text/3-entrees/fixed-point.md

```diff
@@ -0,0 +1,327 @@
+# Fixed Point Arithmetic
+*[`pallets/fixed-point`](https://github.com/substrate-developer-hub/recipes/tree/master/pallets/fixed-point)* *[`pallets/compounding-interest`](https://github.com/substrate-developer-hub/recipes/tree/master/pallets/compounding-interest)*
+
+When programmers learn to use non-integer numbers in their programs, they are usually taught to use [floating point](https://en.wikipedia.org/wiki/Floating-point_arithmetic)s. In blockchain, we use an alternative representation of fractional numbers called [fixed point](https://en.wikipedia.org/wiki/Fixed-point_arithmetic). There are several ways to use fixed point numbers, and this recipe will introduce three of them. In particular we'll see:
+
+* Substrate's own fixed point structs and traits
+* The [substrate-fixed](https://github.com/encointer/substrate-fixed/) library
+* A manual fixed point implementation (and why it's nicer to use a library)
+* A comparison of the two libraries in a compounding interest example
+
+## What's Wrong with Floats?
+
+Floats are cool for all kinds of reasons, but they also have one important drawback. Floating point arithmetic is **nondeterministic**. In order for the nodes in a blockchain network to reach agreement on the state of the chain, all operations must be completely deterministic. Luckily fixed point arithmetic is deterministic, and is often not much harder to use once you get the hang of it.
```

Comment:
We are planning to get a larger doc on DevHub in the next week or two about the problem with floating point and a summary of the tools available. Will definitely ask for you help there too if you don't mind. I think it would be better to dive deep there as Recipes tends to be less conceptual and jumps right into code. Like, "I already know that I want to do X, show me a pattern." 

Also happy to put more here as Joshy sees fit, just my two cents.

Reply:
I've added a tiny bit of details (only as much as @brenzi) mentioned in his comment. I'll leave the deep dive for the devhub knowledge base.